# assignment on measures of central tendancy and dispersion


In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
data=pd.read_csv("I:\ddd\S.csv")

In [3]:
data

gender race/ethnicity parental level of education         lunch  \
0    female        group C            some high school  free/reduced   
1    female        group B                 high school  free/reduced   
2    female        group B            some high school  free/reduced   
3    female        group B                some college      standard   
4    female        group C                some college  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
0                      none           0             17             10  
1                      none           8             24             23  
2                      none          18             32             28  
3                      none          11             38             32  
4                      none          22             39             33  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[1000 rows x 8 columns]

In [4]:
data.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

# 1 count of males and females participaed in the test

In [5]:
count=data['gender'].value_counts()
count

female    518
male      482
Name: gender, dtype: int64

518 females participated in test and 482 males participated in test

# 2 students parental level of education


In [6]:
data['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

226 parents has college level of education,
222 parents has associate degree.
out of 226 parents has some college education,118 parents has bachelors degree and 59 has masters degree

# 3 who scores the most on average for math,reading and writing based on

# a)gender

In [7]:
data1=data[['gender','math score','reading score','writing score']].groupby('gender').mean()

In [8]:
data1

math score  reading score  writing score
gender                                          
female   63.376448      72.590734      72.467181
male     68.821577      65.545643      63.446058

 female has highest average in reading score and writing score whereas   male has highest average in maths

# b) Test preparation course

In [9]:
data2=data[['test preparation course','math score','reading score','writing score']].groupby('test preparation course').mean()


In [10]:
data2

math score  reading score  writing score
test preparation course                                          
completed                  69.96648      74.175978      74.684358
none                       63.78972      66.417445      64.457944

The person who completed test preparation course has highest average in math.reading,and writing compared to the person who does not take test preparation course.

# 4.the scoring variation for maths,reading and writing based on

# a)gender

In [11]:
scor_variation =data[['gender','math score','reading score','writing score']].groupby('gender').std()

In [12]:
scor_variation

math score  reading score  writing score
gender                                          
female   16.029928      14.411018      14.844842
male     14.556411      14.149594      14.227225

The score variation for math,reading and writing is less among males while considering each subjects. females has higher score for each subject when compared with male

# b)test preparation course

In [13]:
scor_variation1=data[['test preparation course','math score','reading score','writing score']].groupby('test preparation course').std()

In [14]:
scor_variation1

math score  reading score  writing score
test preparation course                                          
completed                 14.521847      13.537572      13.236412
none                      15.705689      14.608896      15.041667

The person completed test preparation course has less values  in score compared to one with none of preparation

# 5.top 25% student based on math score

In [15]:
data.shape

(1000, 8)

In [16]:
data.nlargest(250,['math score'])

gender race/ethnicity parental level of education         lunch  \
515  female        group E                some college      standard   
516  female        group E           bachelor's degree      standard   
517  female        group E          associate's degree      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
..      ...            ...                         ...           ...   
428  female        group D                some college  free/reduced   
429  female        group E            some high school      standard   
849    male        group A           bachelor's degree      standard   
850    male        group C          associate's degree  free/reduced   
851    male        group D                some college      standard   

    test preparation course  math score  reading score  writing score  
515                    none         100             92             97  
516                    none         100            100            100  
517                    none         100            100            100  
996               completed         100             96             86  
997               completed         100             97             99  
..                      ...         ...            ...            ...  
428                    none          77             86             86  
429                    none          77             79             80  
849                    none          77             67             68  
850                    none          77             67             64  
851               completed          77             62             62  

[250 rows x 8 columns]

from total of 1000 students ,25% of total students ie 250 students are eligible for bonus points based on the math score